src/test/java/com/facebook/presto/hadoop/TestLineReader.java

```diff
@@ -0,0 +1,83 @@
+package com.facebook.presto.hadoop;
+
+import org.apache.hadoop.io.Text;
+import org.apache.hadoop.util.LineReader;
+import org.testng.annotations.Test;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static org.testng.Assert.assertEquals;
+
+public class TestLineReader {
+
+    @Test
+    public void testDefaultReaderMaxBytesConsumed()
+    {
+        byte[] input = "Hello world! Goodbye world!\n".getBytes(UTF_8);
+        InputStream in = new ByteArrayInputStream(input);
+        // Set the LineReader internal read buffer size 4 bytes
+        LineReader reader = new LineReader(in, 4);
+        Text str = new Text();
+        try {
+            reader.readLine(str, 0, 10);
+        }
+        catch (IOException e) {
+            // It should be 3 reads of 4 bytes each, so the final bytesConsumed is 12
+            assertEquals(e.getMessage(), "Too many bytes before newline: " + 12);
```

Comment:
Let's inline the string concat to make it easier to read:
```java
assertEquals(e.getMessage(), "Too many bytes before newline: 12");
```
Same goes for the other tests.